tests/unit/test_settings_loader.py

```diff
@@ -55,24 +51,22 @@ def test_is_use_default_settings(self):
             self.assertFalse(settings_loader.is_use_default_settings({'use_default_settings': 0}))
 
     def test_user_settings_not_found(self):
-        with patch.dict(settings_loader.environ, {'SEARXNG_SETTINGS_PATH': '/dev/null'}):
-            settings, msg = settings_loader.load_settings()
-            self.assertTrue(msg.startswith('load the default settings from'))
-            self.assertEqual(settings['server']['secret_key'], "ultrasecretkey")
+        with patch.dict(os.environ, {'SEARXNG_SETTINGS_PATH': _settings("not_exists.yml")}):
+            with self.assertRaises(EnvironmentError):
+                _s, _m = settings_loader.load_settings()
+        with patch.dict(os.environ, {'SEARXNG_SETTINGS_PATH': "/folder/not/exists"}):
+            with self.assertRaises(EnvironmentError):
+                _s, _m = settings_loader.load_settings()
```

Comment:
> [!IMPORTANT] 
> **intended functional changes!**
>
> If environment `SEARXNG_SETTINGS_PATH` was set and pointed to a not existing file, the previous implementation silently loaded the default configuration.  This behavior has been changed: if the file or folder does not exist, an EnvironmentError exception will be thrown in future.

Reply:
Not really needed .. its just here in sense of testing "folder not exists" while the first test is in sense of "file not exists".  If you think its superfluous/superfluous I can remove the second test.

Reply:
It's more about the fact that the test fail if `/folder/not/exists` exists since this is folder outside the scope of the repository.

Reply:
When the folder exists, the test will fail as expected:

```
$ sudo mkdir -p /folder/not/exists
$ make test.unit
...
======================================================================
FAIL: test_user_settings_not_found (tests.unit.test_settings_loader.TestUserSettings.test_user_settings_not_found)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests/unit/test_settings_loader.py", line 58, in test_user_settings_not_found
    with self.assertRaises(EnvironmentError):
AssertionError: OSError not raised
```

> since this is folder outside the scope of the repository 

The probability that this absolute path name exists is low, so I would leave it as it is. Nevertheless, thank you for looking at this with a strict eye :+1: